### PR TITLE
Consolidate Gemfiles; improve and simplify CI workflow

### LINKED ISSUE #1000: CI: Failures are prohibitively difficult to troubleshoot
The Cucumber and RSpec steps in CI redirect stderr to a temp file. This hides the output from the GitHub Actions console. As a result, when test execution fails (as recently happened in #995), it is impossible to see what is going wrong. Troubleshooting becomes a much more drawn out process.

E.g.

```
bundle exec rspec spec/ 2>> "${ALL_WARNINGS}"
```

The stderr capture is to facilitate the "check warnings" step, which fails the CI build if any tests output warnings. I assume this is to check for deprecations, so that these can be caught and corrected before impacting end users.

However, in the grand scheme, a deprecation warning is usually more of annoyance than a bug to end users. IMO avoiding these is not worth the trouble of complicating CI.

Perhaps a better way to guard against deprecations that are likely to eventually impact end-users is to add `ruby-head` to the build matrix. That way if a deprecation in a current version of Ruby is planned on becoming an error in a future version, it will fail the ruby-head build and maintainers will have a chance to correct it.

Alternatively, something like the [upload-artifact](https://github.com/actions/upload-artifact) action could be used to upload the captured stderr for troubleshooting purposes. This adds more complexity to the CI job, but if it works reliably, it could make troubleshooting possible while still maintaining the "check warnings" step.

### PR DESCRIPTION
**Pull request by mattbrictson**
The goal of this PR is to improve the long-term maintainability of vcr by greatly simplifying the CI infrastructure and improving its troubleshooting output.

### Improving troubleshooting output

Before, cucumber and rspec commands had their stderr output redirected to a temporary file. The contents of this file were never explicitly printed or archived. If the command failed (as recently happened in #995), maintainers would be flying blind in terms of understanding the failure.

With the changes of this PR, the redirection of stderr still exists, but if a command fails, the stderr output is explicitly printed to the console prior to the CI job exiting. This allows maintainers to see what went wrong in failure scenarios.

Fixes #1000.

### Simplifying Gemfiles and CI infrastructure

Before, the CI workflow was organized as a 1:1 conversion of a previous Travis CI setup, which was structured as one big CI script and multiple Gemfiles.

Over the years, the vcr project dropped the `appraisal` gem in favor of manually managing Gemfile variants. The project also at times tested JRuby and Rubinius, before removing them from CI. These changes and experiments have left a lot of cruft in the testing infrastructure.

This PR performs several cleanup operations:

1. Removes extra complexity of supporting JRuby and Rubinius. These runtimes were not being tested, so the complexity did not serve a purpose.
2. Removes separate Gemfiles. The `Gemfile.faraday-1.0.0` file, for example, had an overlapping version specification with the one in `vcr.gemspec`, effectively making it redundant. In the past, CI was configured to run against different major versions of some key dependencies, but that is no longer the case. Now the multiple Gemfiles do not have any benefit. This PR merges them all into a single Gemfile.
3. Splits the rspec, cucumber, and yard CI steps into parallel jobs. This is something that is easy to do in GitHub Actions that was probably prohibitively difficult in TravisCI. The parallel jobs increase performance and have other benefits.
4. Moves the stderr output redirection and warning detection logic out of the CI workflow and into a dedicated bash script.

My hope is that with fewer Gemfiles to maintain, and a more familiar GitHub Actions workflow, this PR will make vcr more approachable to maintainers and contributors going forward.

### CODE REVIEW

File: vcr.gemspec [hunk 1]
```diff
@@ -20,25 +20,4 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.required_ruby_version = ">= 2.7"
-
-  spec.add_development_dependency "bundler", "~> 2.0"
```
Comment (mattbrictson) [inline vcr.gemspec:24]: 🗒️ These dependencies are now declared in the Gemfile.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,69 +1,56 @@
...
+      - name: rspec
+        run: ./script/fail_if_warnings rspec spec/
+  doc-coverage:
+    name: "Doc coverage"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "ruby"
```
Comment (mattbrictson) [inline .github/workflows/test.yml:53]: 🗒️ I assumed that running the yard coverage check against a matrix of multiple versions of Ruby was overkill. I changed it to run on the latest version of Ruby only.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,11 +16,3 @@ bundle exec rake
 VCR uses [RSpec](http://github.com/rspec/rspec) for unit tests.  The specs are written in a very "focused" style, where each spec is concerned only with exercising the object under test, using mocks as necessary.  You can run the specs using `rake spec`.
 
 [Cucumber](http://cukes.info/) is used for end-to-end full stack integration tests that also function as VCR's documentation.
-
-## Problems running bundle install?
-
-If you get an error while running `bundle install`, it may be one of the "extras" gems which are not required for development. Try installing it without these gems.
-
-```console
-bundle install --without extras
```
Comment (mattbrictson) [inline CONTRIBUTING.md:25]: 🗒️ There is no `extras` group in the Gemfile, so this documentation was misleading. I removed it to prevent confusion.

**Review by olleolleolle — APPROVED** [review]
Thank you, @mattbrictson! I agree with all the changes, and thanks also for all the context-preserving notes in the Description and comments of this PR.

**Pull request merged**